### PR TITLE
ENH: Enable tours to be generated from_geopandas

### LINKED ISSUE #98: tests (and test data) missing for (read) tours
function tours_from_gpd is not tested (#93) 
function read_tours_csv is not available. is this correct or should it be added?

### PR DESCRIPTION
**Pull request by bifbof**
Related to #334 
Additionally I've redone the `_trackintel_model` to handle DataFrames as well as GeoDataFrames. &rarr; Simplified the code for tours and trips.

Edit: 
closes #98

### CODE REVIEW

**Comment by bifbof** [comment]
Is there a way to check which black version the lint-check uses?

**Comment by henrymartin1** [comment]
> 
> 
> Is there a way to check which black version the lint-check uses?

only kind of... There is no version defined in the [workflow](https://github.com/mie-lab/trackintel/blob/master/.github/workflows/black.yml) so I guess it uses the latest (or any random version;-) ). You can check the install log when the action fails. It is a bit hidden though...

https://github.com/mie-lab/trackintel/runs/3620136674?check_suite_focus=true
under _Build lgeiger/black-action@v1.0.1_ [here](https://github.com/mie-lab/trackintel/runs/3620136674?check_suite_focus=true#step:2:62) 
```
  Collecting black
    Downloading black-21.9b0-py3-none-any.whl (148 kB)
```

**Comment by bifbof** [comment]
> > Is there a way to check which black version the lint-check uses?
> 
> only kind of... There is no version defined in the [workflow](https://github.com/mie-lab/trackintel/blob/master/.github/workflows/black.yml?rgh-link-date=2021-09-16T10%3A38%3A21Z) so I guess it uses the latest (or any random version;-) ). You can check the install log when the action fails. It is a bit hidden though...
> 
> [mie-lab/trackintel/runs/3620136674?check_suite_focus=true](https://github.com/mie-lab/trackintel/runs/3620136674?check_suite_focus=true)
> under _Build lgeiger/black-action@v1.0.1_ [here](https://github.com/mie-lab/trackintel/runs/3620136674?check_suite_focus=true#step:2:62)
> 
> ```
>   Collecting black
>     Downloading black-21.9b0-py3-none-any.whl (148 kB)
> ```

Thanks a lot! That helped.

File: tests/io/test_from_geopandas.py [hunk 1]
```diff
@@ -297,9 +297,40 @@ def test_mapper(self, example_trips):
...
+    h = pd.Timedelta("1h")
+
+    list_dict = [
+        {"user_id": 0, "started_at": start + 0 * h, "finished_at": start + 1 * h},
+        {"user_id": 0, "started_at": start + 2 * h, "finished_at": start + 3 * h},
+        {"user_id": 1, "started_at": start + 4 * h, "finished_at": start + 5 * h},
+    ]
+    tours = pd.DataFrame(data=list_dict)
+    tours.index.name = "id"
+    assert tours.as_tours
```
Comment (hong2223) [inline tests/io/test_from_geopandas.py:313]:
```suggestion
    tours.as_tours
```

**Review by hong2223 — APPROVED** [review]

**Comment by hong2223** [comment]
Nice! I only changed the assert statement, this should be ready for merging

**Pull request merged**